src/controllers/plugins/PluginPermission.cpp

```diff
@@ -0,0 +1,45 @@
+#ifdef CHATTERINO_HAVE_PLUGINS
+#    include "controllers/plugins/PluginPermission.hpp"
+
+#    include <magic_enum/magic_enum.hpp>
+#    include <QJsonObject>
+
+namespace chatterino {
+
+PluginPermission::PluginPermission(const QJsonObject &obj)
+{
+    auto jsontype = obj.value("type");
+    if (!jsontype.isString())
+    {
+        QString tn = magic_enum::enum_name(jsontype.type()).data();
+        this->errors.emplace_back(QString("permission type is defined but is "
+                                          "not a string (its type is %1)")
+                                      .arg(tn));
+    }
+    auto strtype = jsontype.toString().toStdString();
+    auto opt = magic_enum::enum_cast<PluginPermission::Type>(
+        strtype, magic_enum::case_insensitive);
+    if (!opt.has_value())
+    {
+        this->errors.emplace_back(QString("permission type is an unknown (%1)")
+                                      .arg(jsontype.toString()));
+        return;  // There is no more data to get, we don't know what to do
+    }
+    this->type = opt.value();
+}
+
+QString PluginPermission::toHtmlEscaped() const
+{
+    switch (this->type)
+    {
+        case PluginPermission::Type::FilesystemRead:
+            return "In its data directory.";
```

Comment:
This reads as `Required permissions: In its data directory`. Is this intended?

Reply:
I think I wrote this at 3 in the morning. I'm happy to reword it.